File: SMS Multicast/app/src/main/java/org/yuzhiqiang/smsmulticast/Message.java
package org.yuzhiqiang.smsmulticast;

import java.io.Serializable;
import java.util.UUID;

public class Message implements Serializable {
	public static final int NEW = 0;
	public static final int PENDING = 1;
	public static final int SENDING = 2;
	public static final int SENT = 3;
	public static final int FAILED = -1;
	private final UUID uuid;
	public String destination;
	public String content;
	public int status;

	public Message(String destination, String content) {
		this.uuid = UUID.randomUUID();
		this.destination = destination;
		this.content = content;
		this.status = NEW;
	}

	@Override
	public boolean equals(Object o) {
		return (((Message) o).uuid.equals(this.uuid));
	}

	public boolean isSentOut() {
		return (status > 0);
	}
}

File: SMS Multicast/app/src/main/java/org/yuzhiqiang/smsmulticast/Message.kt
package org.yuzhiqiang.smsmulticast

import java.io.Serializable
import java.util.*

class Message(var destination: String, var content: String) : Serializable {
    private val uuid: UUID
    var status: Int = 0

    val isSentOut: Boolean
        get() = status > 0

    init {
        this.uuid = UUID.randomUUID()
        this.status = NEW
    }

    override fun equals(o: Any?): Boolean {
        return (o as Message).uuid == this.uuid
    }

    companion object {
        val NEW = 0
        val PENDING = 1
        val SENDING = 2
        val SENT = 3
        val FAILED = -1
    }
}
